### DISCUSSION TITLE
Progress bar metrics

### DISCUSSION THREAD

**Discussion by stalevna** [NVIDIA-NeMo/NeMo#9373] [Q&A]
Hi everyone!
I want to learn how to change progress bar metrics when training. I used to have loss showing in the bar, but now I have train_step_timing which is not very useful. Could you help me figure out where I can change these settings? I tried to search in the Pytorch files, but the places where you change it were commented out.
This is the example of how it shows right now:
32:29<5:19:08,  4.81it/s, v_num=0-12, train_step_timing in s=0.122

**Comment by titu1994** [comment]
It depends on the model, but in general were logging the results in a models training_step() and validation_step()

**Reply by stalevna** [reply]
it is an ASR model. I have checked and the training_step() returns loss and that is exactly what I need but for some reason in the progress bar I still get train_step_timing. I will try to install and put RichProgressBar in callbacks -> trainer = Trainer(callbacks=RichProgressBar()) and see if it helps

**Reply by stalevna** [reply]
Unfortunately, RichProgressBar didn't help
![image](https://github.com/NVIDIA/NeMo/assets/53433879/acc9981a-3e31-42f9-a770-0599158270bb)
Going to look into pytorch_lightning, maybe logging results are overwritten somewhere
